content/50_deploying_argocd_to_eks/30_declare_argocd_helm_chart.md

```diff
@@ -0,0 +1,98 @@
++++
+title = "Declare an ArgoCD Helm Chart Object"
+chapter = false
+weight = 30
++++
+
+You'll now declare a [helm chart](https://kubernetes.io/blog/2016/10/helm-charts-making-it-simple-to-package-and-deploy-apps-on-kubernetes/),
+which deploys all of the components that is required to deploy ArgoCD.
+
+Next, let's declare the helm chart:
+
+```typescript
+const argocd = new k8s.helm.v2.Chart("argocd",
+    {
+        namespace: ns.metadata.name,
+        chart: "argo-cd",
+        fetchOpts: { repo: "https://argoproj.github.io/argo-helm" },
+        values: {
+            installCRDs: false,
+            server: {
+                service: {
+                    type: 'LoadBalancer',
+                },
+            }
+        },
+        // The helm chart is using a deprecated apiVersion,
+        // So let's transform it
+        transformations: [
+            (obj: any) => {
+                if (obj.apiVersion == "extensions/v1beta1")  {
```

Comment:
Should also check against the `obj.kind` (guessing it's "Ingress")

Reply:
if the API is deprecated, doesn't it make sense to migrate all the kind's to the new one?

Reply:
Some of the `extensions` resources moved to different apiVersions (`apps`, `networking`, `rbac`), so this could generate invalid resources depending on what's in the chart.

Reply:
@jaxxstorm feel free to push to the branch with what you want to update here ;)